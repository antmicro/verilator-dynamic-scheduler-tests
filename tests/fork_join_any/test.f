$TEST_DIR/test.sv $REPO_DIR/src/main_new.cpp
