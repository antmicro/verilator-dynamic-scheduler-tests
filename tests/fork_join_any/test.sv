module t (/*AUTOARG*/
   // Inputs
   clk
   );
   input clk;
   event cont;

   initial begin
      fork
         $write("forked process 1\n");
         begin
            @cont;
            $write("forked process 2\n");
            $write("*-* All Finished *-*\n");
            $finish;
         end
      join_any
      #1;
      $write("main process\n");
      ->cont;
   end
endmodule
