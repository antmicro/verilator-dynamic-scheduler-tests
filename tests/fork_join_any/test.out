forked process 1
main process
forked process 2
*-* All Finished *-*
